tests/python/relax/test_op_gradient_numeric.py

```diff
@@ -0,0 +1,408 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import numpy as np
+import pytest
+import tvm
+from tvm import relax
+from tvm.relax.transform import LegalizeOps
+from tvm.testing.utils import check_numerical_grads
+from tvm.ir.op import Op
+from typing import Callable, Union, Tuple, List
+
+
+def relax_check_gradients(
+    op_func: Callable,
+    op_name: str,
+    inputs_numpy: np.array,
+    target: Union[str, tvm.target.Target],
+    dev: tvm._ffi.runtime_ctypes.Device,
+    output_shape: Union[Tuple, List[Tuple]],
+    tuple_input: bool = False,
+    **kwargs,  # attr for operators
+):
+    """Generate module and run it to check numberic gradients."""
+
+    func_name = "main"
+
+    # prepare input
+    def _numpy_to_var(data, var_name):
+        if isinstance(data, list):
+            struct_infos = []
+            for i in range(len(data)):
+                tvm_var = _numpy_to_var(data[i], "")
+                struct_infos.append(tvm_var.struct_info)
+            return relax.Var(var_name, relax.TupleStructInfo(struct_infos))
+        return relax.Var(var_name, relax.TensorStructInfo(data.shape, "float32"))
+
+    def _numpy_to_tvm(data):
+        if isinstance(data, list):
+            ret_data = []
+            for i in range(len(data)):
+                tvm_data = _numpy_to_tvm(data[i])
+                ret_data.append(tvm_data)
+            return tvm.runtime.container.ADT(0, ret_data)
+        return tvm.nd.array(data)
+
+    def _tvm_to_numpy(data):
+        if isinstance(data, tvm.runtime.container.ADT):
+            return [_tvm_to_numpy(i) for i in data]
+        return data.numpy()
+
+    def _gen_weights(shape):
+        if isinstance(shape, list):
+            ret = []
+            for s in shape:
+                ret.append(_gen_weights(s))
+            return ret
+        else:
+            return np.random.uniform(size=shape).astype(np.float32)
+
+    param_vars = [_numpy_to_var(inputs_numpy[i], "x_" + str(i)) for i in range(len(inputs_numpy))]
```

Comment:
I think `enumerate` is helpful here.
```suggestion
    param_vars = [_numpy_to_var(input_numpy, "x_" + str(i)) for i, input_numpy in enumerate(inputs_numpy)]
```